node-notes-cli/read.js

```diff
@@ -0,0 +1,15 @@
+import { readFile } from 'node:fs/promises';
+
+// const note = process.argv[2];
+
+(async () => {
+  try {
+    // Read the existing JSON data from 'data.json'
+    const existingData = await readFile('data.json', 'utf8');
+    const data = JSON.parse(existingData);
+    console.log(data);
+    // error handling
+  } catch (error) {
+    console.error('Error:', error);
+  }
+})();
```

Comment:
This approach where you declare the function and then call it immediately is an old way of writing the code, which isn't necessary in modern JavaScript. The version of node we have lets you just write the try/catch directly without the surrounding function definition and call.